frontend/src/js/components/Canvas.jsx

```diff
@@ -20,25 +20,36 @@ var Canvas = React.createClass({
         canvas.attr("width", div.width());
         canvas.attr("height", div.height());
         this.canvas.clear()
-        this.props.plots.map((plot, index) => {
-            if (plot.variables.length > 0) {
-                var variables = this.props.plotVariables[index];
-                var dataSpecs = variables.map(function (variable) {
-                    var dataSpec = Object.assign({}, variable.provenance);
-                    var subRegion = {};
-                    Object.values(variable.dimension)
-                        .filter(dimension => dimension.range)
-                        .forEach((dimension) => {
-                            subRegion[dimension.axisName] = dimension.range;
-                        })
-                    if (!_.isEmpty(subRegion)) {
-                        dataSpec.operations = [{ subRegion }];
-                    }
-                    return dataSpec;
-                });
-                this.canvas.plot(dataSpecs, this.props.plotGMs[index], plot.template);
-            }
-        });
+        if (this.props.plots !== prevProps.plots) {
+            this.props.plots.map((plot, index) => {
+                if (plot.variables.length > 0) {
+                    var variables = this.props.plotVariables[index];
+                    var dataSpecs = variables.map(function (variable) {
+                        var dataSpec = {
+                            uri: variable.path,
+                            variable: variable.cdms_var_name
+                        };
+                        var subRegion = {};
+                        variable.dimension
+                            .filter(dimension => dimension.values)
+                            .forEach((dimension) => {
+                                subRegion[dimension.axisName] = dimension.values.range;
+                            })
+                        if (!_.isEmpty(subRegion)) {
+                            dataSpec['operations'] = [{ subRegion }];
+                        }
+
+                        var axis_order = variable.dimension.map((dimension) => variable.axisList.indexOf(dimension.axisName));
+                        if (axis_order.some((order, index) => order !== index)) {
+                            dataSpec['axis_order'] = axis_order;
+                        }
+                        return dataSpec;
+                    });
+                    console.log('plotting', dataSpecs, this.props.plotGMs[index], plot.template);
```

Comment:
do we need this, or is this testing/debugging code? I'm a big fan of no debug / print statements in the repo.

Reply:
This is everything got sent to vcs.js for plotting. There are many plottings not working correctly, and the plotting API is changing rapidly at this moment. So I would like to have this committed so I don't need to add it every time. I'll definitely remove all of them at certain time.